SomerenUI/EditDrinkForm.cs

```diff
@@ -31,40 +29,38 @@ private void btnCreateDrink_Click(object sender, EventArgs e)
             bool isAlcoholic = rdoTrue.Checked;
             try
             {
-                if (txtDrinkName.Text.IsNullOrEmpty())
-                {
-                    throw new Exception("Please input a name for the drink!");
-                }
+                if (txtDrinkName.Text.IsNullOrEmpty())                
+                    throw new ArgumentNullException();                
 
                 Drink drink = new Drink(double.Parse(txtDrinkPrice.Text), isAlcoholic, txtDrinkName.Text, int.Parse(txtDrinkStock.Text));
```

Comment:
I suggest breaking down operations into more digestible steps here.
Splitting operations is a widely accepted coding practice, especially regarding readability and maintainability. Breaking down operations into more digestible steps, particularly when it involves parsing or converting data types, can significantly enhance clarity and ease of debugging.